clippy_lints/src/empty_docs.rs

```diff
@@ -0,0 +1,151 @@
+use itertools::Itertools;
+use rustc_ast::ast;
+use rustc_ast::ast::{Attribute, Block, Item};
+use rustc_ast::token;
+use rustc_ast::tokenstream;
+use rustc_ast::AttrKind::{DocComment, Normal};
+use rustc_lint::{EarlyContext, EarlyLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::BytePos;
+use rustc_span::Span;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for empty documentation
+    ///
+    /// ### Why is this bad?
+    /// It is unlikely there is any reason to have empty documentation for an entity
+    ///
+    /// ### Example
+    /// ```rust
+    /// ///
+    /// fn returns_true() {
+    ///     true
+    /// }
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// fn returns_true() {
+    ///     true
+    /// }
+    /// ```
+    #[clippy::version = "1.68.0"]
+    pub EMPTY_DOCS,
+    suspicious,
+    "docstrings exist but documentation is empty"
+}
+declare_lint_pass!(EmptyDocs => [EMPTY_DOCS]);
+
+const MAX_CONTEXT_LEN: BytePos = BytePos(50);
+
+impl EarlyLintPass for EmptyDocs {
+    fn check_item(&mut self, ex: &EarlyContext<'_>, item: &Item) {
+        if item.span.from_expansion() {
+            return;
+        }
+        self.process_attributes(ex, item.span, &item.attrs.to_vec());
+        self.process_into_item(ex, &item);
+    }
+}
+
+impl EmptyDocs {
+    fn process_into_item(self, ex: &EarlyContext<'_>, item: &Item) {
+        match &item.kind {
+            ast::ItemKind::Struct(ast::VariantData::Struct(field_list, _), _) => {
+                for field_def in field_list {
+                    self.process_attributes(ex, field_def.span, &field_def.attrs.to_vec());
+                }
+            },
+            ast::ItemKind::Union(ast::VariantData::Struct(field_list, _), _) => {
+                for field_def in field_list {
+                    self.process_attributes(ex, field_def.span, &field_def.attrs.to_vec());
+                }
+            },
+            ast::ItemKind::Enum(enum_def, _) => {
+                for variant in &enum_def.variants {
+                    self.process_attributes(ex, variant.span, &variant.attrs.to_vec());
+                }
+            },
+            ast::ItemKind::Fn(fn_def) => {
+                if let Some(block) = &fn_def.body {
+                    self.process_into_block(ex, &block);
+                }
+            },
+            _ => {},
+        }
+    }
+
+    fn process_into_block(self, ex: &EarlyContext<'_>, block: &Block) {
+        for statement in &block.stmts {
+            let statement_span = statement.span;
+            match &statement.kind {
+                ast::StmtKind::Local(local) => {
+                    self.process_attributes(ex, statement_span, &local.attrs.to_vec());
+                },
+                _ => {},
+            }
+        }
+    }
+
+    fn process_attributes(self, ex: &EarlyContext<'_>, parent_span: Span, attributes: &Vec<Attribute>) {
```

Comment:
We have some code that reassembles doc comments for some related lints, the lint could be moved into `doc.rs` and check [here-ish](https://github.com/rust-lang/rust-clippy/blob/decaba97cc0e682a7b44b3f06f10ad1a8ebe9cba/clippy_lints/src/doc.rs#L536) based on `spans` + `doc`

Sorry to point that out after you've done the complicated part yourself already 😬

Not the first time this has happened with `doc.rs`, maybe it would be worth renaming it to `doc_comments.rs` or something, but it'll still be easy to miss